<a href="https://colab.research.google.com/github/sflmnt/Business-Intelligence/blob/main/Reto2_Sof%C3%ADaLomanto.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Reto 2:** Sofía Lomanto

**0.** Se procede con la carga de las librerias de trabajo

In [ ]:
#Librerias Cientificas
import numpy as np
import pandas as pd

#Librerias Arbol
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import confusion_matrix

#Conectar al Drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


**1.** Se procede con la carga de los datos

In [ ]:
nxl='/content/drive/MyDrive/Análitica 2025-1/2. CollegeAdmisions_Int_M.xlsx'
XDB=pd.read_excel(nxl,sheet_name=0) #Primera hoja
XDB.head(100)

#Seleccionar las variables
XD=XDB[['Edu_Parent1','Edu_Parent2','Gender','White','Asian','HSGPA','SAT/ACT']]
yd=XDB['Admitted']

# **Descripción de los datos:**
De acuerdo a la base de datos de estudiantes que son admitidos o no a la universidad, esta posee un total de 17339 datos totales, divididos en 5323 admitidos y 12016 rechazados. Cada uno de los aspirantes estará descrito en función de 7 variables:

* **Education Parent 1 y Education Parent 2:** Indica el nivel educativo del primer y segundo padre/madre del estudiante, con los  valores de **1** (Sin escuela preparatoria) , **2** (Escuela preparatoria trunca), **3** (Graduado/a de la escuela preparatoria), **4** (Universidad trunca), **5** (No especificado), **6** (Graduado universitario en carrera de 4 años) y **7** (Posgrado).
* **Gender:** Muestra el género del solicitante: M (1) para hombres y F (0) para mujeres.
* **White:** Señala si el solicitante se identifica como blanco: (1) para blanco
 y (0) para no blanco.
* **Asian:** Señala si el solicitante se identifica como asiático: (1) para asiático y (0) para no asiático.
* **HSGPA:** Nos muestra el promedio académico (GPA) ponderado de la escuela preparatoria del solicitante, en una escala de 0 a 5.
* **SAT/ACT:** Representa la puntuación más alta obtenida en las pruebas estandarizadas SAT o ACT.La puntuación del ACT se convierte primero a un equivalente en el SAT, tomando en cuenta las secciones de inglés y matemáticas.
* **Admitted:** Indica si el estudiante fue admitido o no a la universidad: (1) para admitidos y (0) para no admitidos.


**Nota:** información sacada del libro de la materia "ANALÍTICA DE NEGOCIOS 2ED: Comunicación con números" de los autores Alison Kelly , Kevin Lertwachara , Leida Chen y Sanjiv Jaggia.

**2.** Se procede con la implementación del modelo

In [ ]:
mar=DecisionTreeClassifier(criterion='gini',max_depth=4)
mar.fit(XD,yd)

#Se procede con la gráfica
from pydotplus import graph_from_dot_data #Esto crea la gráfica
from sklearn.tree import export_graphviz #Esto la exporta a PNG, JPEG, TIFF

ve= ['Edu_Parent1','Edu_Parent2','Gender','White','Asian','HSGPA','SAT/ACT'] #Nombres para el árbol
dot_data=export_graphviz(mar,feature_names=ve) #Crear el gráfico
graph= graph_from_dot_data(dot_data)
graph.write_png('2. ArbolDecisión_Reto 2.png') #Exportar a PNG


True

**3.** Se procede con el pronóstico de la admisión para las personas agrupadas en la base de datos de pronóstico.

In [ ]:
XDB0=pd.read_excel(nxl,sheet_name=1)
XDB0.head(100)

#Se crea la base de datos a pronósticar
XD0=XDB0[['Edu_Parent1','Edu_Parent2','Gender','White','Asian','HSGPA','SAT/ACT']]
yd0=XDB0['Admitted']

#Le calculamos la admisión a los estudiantes
y_pred= mar.predict(XD0)
print(y_pred)

#Se crea el informe en excel
df=pd.DataFrame(np.column_stack((XD0,y_pred)))
df.columns=['Edu_Parent1','Edu_Parent2','Gender','White','Asian','HSGPA','SAT/ACT','Admitted']
df.to_excel('Evaluación admisión Reto 2.xlsx')


['No' 'Yes' 'No' 'No' 'No' 'No' 'No' 'No' 'No' 'No' 'Yes']


**4.** Se evalúa el comportamiento del modelo

In [ ]:
ydp=mar.predict(XD) #Pronóstico sobre la base de datos original
cm=confusion_matrix(yd,ydp) #Matriz de confusión
print("La matriz de confusión es: \n",cm)

VN=cm[0,0] #Verdaderos Negativos
VP=cm[1,1] #Verdaderos Positivos
FP=cm[0,1] #Falsos Positivos
FN=cm[1,0] #Falsos Negativos

Exactitud=(VP+VN)/(VP+VN+FP+FN)
print("La exactitud es: ",Exactitud)

Sensibilidad=VP/(VP+FN)
print("La sensibilidad es: ",Sensibilidad)

Especificidad=VN/(VN+FP)
print("La especificidad es: ",Especificidad)

Precision=VP/(VP+FP)
print("La precisión es: ",Precision)

TasaError=(FP+FN)/(VP+VN+FP+FN)
print("La tasa de error es:",TasaError)

PredNeg=VN/(VN+FP)
print("La probabilidad de no ser admitido es: ",PredNeg)

La matriz de confusión es: 
 [[10719  1297]
 [ 1985  3338]]
La exactitud es:  0.8107157275506085
La sensibilidad es:  0.627089986849521
La especificidad es:  0.892060585885486
La precisión es:  0.7201725997842503
La tasa de error es: 0.18928427244939156
La probabilidad de no ser admitido es:  0.892060585885486


# **Análisis de Resultados:**
La matriz de confusión muestra que se predominan los Verdaderos Negativos (10719) y los Verdaderos Positivos (3338), con base a estos resultados el modelo demostró ser excelente en cuanto a exactitud y especificidad, estando por encima del 80% (81% y 89,2% respectivamente). También el modelo consiguió una precisión del 72% que demuestra su buen comportamiento general admitiendo estudiantes y finalmente, toda esta información es corroborada por la baja tasa de error de el modelo que se encuentra por debajo del 20% (18,9%) y demuestra su buen comportamiento negando la admisión.

En cuanto a el pronóstico de la admisión para las personas agrupadas en la base de datos de pronóstico, de los 11 aplicantes estudiados, el modelo acepto a dos aplicantes (18,18%) y no dio admisión a los 9 restantes (81.81%) esto se alinea con la probabilidad de no ser admitido que fue en el modelo de un 89%.

Finalmente, en el árbol de decisión del modelo no se encontraron nodos puros, pero se resalta la presencia de algunos nodos con un "gini" especialmente bajo con resultados de 0,042 , 0,092 y 0,094.